articles/ssh-in-nodejs/index.md

```diff
@@ -0,0 +1,149 @@
+
+#### Using SSH
+Most UNIX based operating systems such as Ubuntu, MAC OS, etc. come with an SSH client pre-installed. To establish a connection, open the terminal and execute the following command:
+
+`ssh <username>@<hostname or host ip>`
+
+The username field specifies the user id of the machine you are connecting to. The hostname is the name of the remote system and can be found by typing `hostname` in the terminal. Users can also specify the host IP instead of the hostname. for example:
+
+`ssh adith@219.123.101.201`
+
+This establishes a connection with the host whose IP is: 219.123.101.201 and the user is adith. 
+
+### SSH in NodeJs
+We are going to build an application that connects to a virtual machine and extracts information on the filesystem health of the VM. The application monitors the disk utilization in the VM and flags the partitions that have a disk utilization greater than 90%. This is useful for system administrators as it tells which partitions need to be monitored and increased in size before an actual failure of the system due to low space. 
+
+Node has an array of libraries for establishing an SSH connection. In this article, we are going to use [simple-ssh](https://www.npmjs.com/package/simple-ssh). Simple-ssh is a robust and easy to use library that allows us to connect to a VM and execute Linux commands using Node.
+
+Install the library by running the following command:
+
+`npm install simple-ssh`
+
+### Code
+```node
+// required to take input from user
+const readline = require("readline");
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+```
+
+The `readline` library allows users to provide input using the command line. We are going to take the VM host details such as hostname, username, and password as input from the user. This is required to establish a secure connection with the remote host. 
+
+```node
+// function to get input from user
+function get_input() {
+
+	rl.question("What is the host? ", function(host) {
+	    rl.question("What is the user? ", function(user) {
+	    	rl.question("What is the password? ", function(password) {	
+	    		
+	    		ssh(host, user ,password);
+	   			rl.close();
+	    	
+	    	});
+		});
+	});
+}
+```
+
+the `get_input()` function takes input from the user. The hostname, username, and password of the VM are stored in the `host`, `user`, and `password` variables and are passed to the ssh() function as parameters.
+
+```node
+// function to ssh into a remote host.
+function ssh(host, user ,password){
+
+	console.log('inside the system')
+	var SSH = require('simple-ssh');
+	var ssh = new SSH({
```

Comment:
This doesn't apply if you decide to use `var` instead of `const` for the require assignments, and it's also pretty nit-picky, but if you're going to use `const`, then you might as well use `let` here instead of `var`.